x-team/xp-ui
# Add support for “placeholder” prop

**Release Type:** *Non-Breaking Improvement* 


Fixes https://x-team-internal.atlassian.net/browse/XP-2690

## Description

This is an upgrade of the `tui-editor` lib that should bring no breaking changes. The main goal is to allow support for the `placeholder` prop on the `RichTextEditor`. Nothing more has been done in the scope of the ticket. Just a lib upgrade and adding support for a new prop. All existing app functionality and prop values left the same.

## Checklist

**Before submitting a pull request,** please make sure the following is done:



## Steps to Test or Reproduce

1. I'd recommend upgrading lib with the following steps under `xp-ui` dir in terminal:
    1. run `rm -rf node_modules` to clean up `node_modules` directory
    1. run `npm i` to install all the deps
1. Run Storybook with `npm start` and check the new `with placeholder specified` [story](http://localhost:9001/?path=/story/core-components-form-components-richtexteditor-debug--with-placeholder-specified) on the `RichtextEditor` UI component. Make sure default placeholder is shown and you can type custom text so placeholder disappears and when you clear out custom text, a placeholder is back again.
1. Copy files under `/lib*` folder over to `xp`'s `node_modules/xp-ui/lib/*` folder and restart `xp` project with `npm run watch`
1. Run `XP` app and make some local tests:
    1. Check `Notes` section on the applicant's profile in admin. Make sure you can still type note text with different formatting and can editing existing notes with correctly rendered formatting.
    1. Check `Profile Summary`, `Experience Examples`, `Unleash Goals`, and `Favorite Interview Question` applicant profile sections in admin. Make sure you can type text and see no visual regressions.
    1. Check `Contact` section on the applicant's profile in admin. Make sure you see `Enter details...` as an initial placeholder for the `Body` field and the `Send` button is disabled. Type smth. into both `Subject` and `Body` and observe the `Submit` button gets enabled. Remove custom body text, and observe that `Enter details...` placeholder is back and the Send button is disabled.

## Impacted Areas in Application

Extended `RichTextEditor` component to support `placeholder` prop.

## Screenshots

#### Placeholder is visible in the Contact UI

![image](https://user-images.githubusercontent.com/579331/71450161-3d34a400-276d-11ea-8f9c-66b7bb5564d8.png)

#### Storybook case

![image](https://user-images.githubusercontent.com/579331/71450221-af59b880-276e-11ea-9654-2d8520d5c6f0.png)
